### ISSUE TITLE
TreeViewState ignores collapsed/expanded?

### ISSUE THREAD

**Issue by jrieken** [Microsoft/vscode#67917]
re #64727

* open a TypeScript file
* see outline and collapse a few nodes
* switch to another file with outline, e.g. json
* switch back to the TypeScript file
* :bug: collapsed nodes are expanded again

@joaomoreno It does tree view states like this but it doesn't seem to honour expansion. https://github.com/Microsoft/vscode/blob/9f59f4054872c4e49157be0825a1410bed5fdcca/src/vs/workbench/parts/outline/electron-browser/outlinePanel2.ts#L489-L490

**Comment by jrieken** [comment]
I believe this is fixed via https://github.com/Microsoft/vscode/issues/67912

**Comment by joaomoreno** [comment]
@jrieken `joh/o2` branch is currently broken to test this:

```
  ERR Cannot read property 'getModel' of undefined: TypeError: Cannot read property 'getModel' of undefined
    at OutlinePanel.<anonymous> (file:///home/joao/Work/vscode/out/vs/workbench/parts/outline/electron-browser/outlinePanel.js:382:48)
    at step (file:///home/joao/Work/vscode/out/vs/workbench/parts/outline/electron-browser/outlinePanel.js:57:23)
    at Object.next (file:///home/joao/Work/vscode/out/vs/workbench/parts/outline/electron-browser/outlinePanel.js:38:53)
    at file:///home/joao/Work/vscode/out/vs/workbench/parts/outline/electron-browser/outlinePanel.js:32:71
    at new Promise (<anonymous>)
    at __awaiter (file:///home/joao/Work/vscode/out/vs/workbench/parts/outline/electron-browser/outlinePanel.js:28:12)
    at OutlinePanel._doUpdate (file:///home/joao/Work/vscode/out/vs/workbench/parts/outline/electron-browser/outlinePanel.js:373:20)
    at RequestOracle._callback (file:///home/joao/Work/vscode/out/vs/workbench/parts/outline/electron-browser/outlinePanel.js:301:142)
    at RequestOracle._update (file:///home/joao/Work/vscode/out/vs/workbench/parts/outline/electron-browser/outlinePanel.js:108:22)
    at RequestOracle (file:///home/joao/Work/vscode/out/vs/workbench/parts/outline/electron-browser/outlinePanel.js:90:18)
    at create (file:///home/joao/Work/vscode/out/vs/base/common/types.js:164:14)
    at InstantiationService._createInstance (file:///home/joao/Work/vscode/out/vs/platform/instantiation/common/instantiationService.js:107:35)
    at InstantiationService.createInstance (file:///home/joao/Work/vscode/out/vs/platform/instantiation/common/instantiationService.js:76:31)
    at file:///home/joao/Work/vscode/out/vs/workbench/parts/outline/electron-browser/outlinePanel.js:301:72
    at Emitter.fire (file:///home/joao/Work/vscode/out/vs/base/common/event.js:542:38)
    at OutlinePanel.ViewletPanel.setVisible (file:///home/joao/Work/vscode/out/vs/workbench/browser/parts/views/panelViewlet.js:56:53)
    at _loop_1 (file:///home/joao/Work/vscode/out/vs/workbench/browser/parts/views/viewsViewlet.js:175:23)
    at ExplorerViewlet.ViewContainerViewlet.onDidAddViews (file:///home/joao/Work/vscode/out/vs/workbench/browser/parts/views/viewsViewlet.js:190:17)
    at ExplorerViewlet.ViewContainerViewlet.create (file:///home/joao/Work/vscode/out/vs/workbench/browser/parts/views/viewsViewlet.js:73:22)
    at ExplorerViewlet.create (file:///home/joao/Work/vscode/out/vs/workbench/parts/files/electron-browser/explorerViewlet.js:141:37)
    at SidebarPart.CompositePart.showComposite (file:///home/joao/Work/vscode/out/vs/workbench/browser/parts/compositePart.js:138:27)
    at SidebarPart.CompositePart.doOpenComposite (file:///home/joao/Work/vscode/out/vs/workbench/browser/parts/compositePart.js:85:18)
    at SidebarPart.CompositePart.openComposite (file:///home/joao/Work/vscode/out/vs/workbench/browser/parts/compositePart.js:57:25)
...
```

**Comment by jrieken** [comment]
pushed a change for that

**Comment by joaomoreno** [comment]
Yeah this seems fixed now, possibly via #67912
